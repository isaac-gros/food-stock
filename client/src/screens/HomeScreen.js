import React from 'react';
import { View, Text, FlatList } from 'react-native';

class HomeScreen extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      data: [
        {
          id: 1,
          name: 'Banane',
          batchs: [
            {
              id: 1,
              quantity: 3,
              created_at: 1612976706,
              expired_at: 1613001906,
              category: {
                id: 1,
                name: 'Fruits',
                average_expiry: 604800,
              },
            },
          ],
        },
        {
          id: 2,
          name: 'Steak',
          batchs: [],
        },
        {
          id: 3,
          name: 'Truite',
          batchs: [
            {
              id: 2,
              quantity: 1,
              created_at: 1612976706,
              expired_at: 1613235906,
              category: {
                id: 1,
                name: 'Poisson',
                average_expiry: 604800,
              },
            },
          ],
        },
      ],
    };
  }

  componentDidMount() {
    // console.log(this.props.data);
  }

  render() {
    const { data } = this.state;

    return (
      <View>
        {data.map(d => (
          <View key={d.id}>
            <Text>{d.name}</Text>
          </View>
        ))}
      </View>
    );
  }
}

export default HomeScreen;
