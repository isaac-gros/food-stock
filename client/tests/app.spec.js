import React from 'react';
import renderer from 'react-test-renderer';
import HomeScreen from '../src/screens/HomeScreen';

/**
 * To remove the obselete test : npm test -- -u
 */


// Test to Renderer JSX
test("Test", ()=> {
	const snap = renderer.create(
		<div><h1>Hello World!</h1></div>
	).toJSON();
	expect(snap).toMatchSnapshot();
});

// Test to renderer the Home screen
describe("Products", () => {
	it('Should return products as props', () => {
		// Pass arr to  props
		const snap = renderer.create(
			<HomeScreen/>
		).toJSON();
		expect(snap).toMatchSnapshot();
	});
});
