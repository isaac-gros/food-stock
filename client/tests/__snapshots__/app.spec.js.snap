// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Products Should return products as props 1`] = `
<View>
  <View>
    <Text>
      Banane
    </Text>
  </View>
  <View>
    <Text>
      Steak
    </Text>
  </View>
  <View>
    <Text>
      Truite
    </Text>
  </View>
</View>
`;

exports[`Test 1`] = `
<div>
  <h1>
    Hello World!
  </h1>
</div>
`;
